src/Compilers/VisualBasic/Portable/Symbols/AssemblySymbol.vb

```diff
@@ -750,8 +750,18 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                 Return True
             End If
 
+            ' Avoid using the identity to obtain the public key if possible to avoid the allocations associated
+            ' with identity creation
+            Dim publicKey As ImmutableArray(Of Byte)
+            If TypeOf assemblyWantingAccess Is AssemblySymbol Then
```

Comment:
>If TypeOf assemblyWantingAccess Is AssemblySymbol Then [](http://example.com/codeflow?start=12&length=54)

I think doing the following instead would be more efficient:
```
Dim assemblyWantingAccessAssemblySymbol As AssemblySymbol = TryCast(assemblyWantingAccess, AssemblySymbol)
If assemblyWantingAccessAssemblySymbol IsNot Nothing Then
```

Reply:
Commit 3. TryCast is the VB call that I tend to forget about.